conftest.py

```diff
@@ -19,3 +26,26 @@ def pytest_addoption(parser):
                      action="store",
                      default="",
                      help="GGL CLI Path")
+
+
+@pytest.fixture(autouse=True)
+def cleanup_after_test():
+    """Cleanup greengrass state after each test to prevent state pollution"""
+    yield    # Test runs here
+    print("\nCleaning up greengrass state after test...")
+    clean_up()
+
+    # Clear systemd journal logs to avoid log pollution between tests
+    try:
+        import subprocess
+        subprocess.run(["journalctl", "--rotate"],
+                       check=False,
+                       capture_output=True,
+                       timeout=5)
+        subprocess.run(["journalctl", "--vacuum-time=1s"],
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Starting a process with a partial executable path https://bandit.readthedocs.io/en/latest/plugins/b607_start_process_with_partial_path.html